modules/UI/recording/Recording.js

```diff
@@ -391,15 +411,15 @@ var Recording = {
 
             this._updateStatusLabel(this.recordingPendingKey, true);
         }
-        else if (recordingState === Status.ERROR) {
+        else if (recordingState === Status.ERROR
+                    || recordingState === Status.FAILED) {
             buttonSelector.removeClass(this.baseClass + " active");
             buttonSelector.addClass(this.baseClass);
 
             this._updateStatusLabel(this.recordingErrorKey, true);
+            console.log("Recording failed for the following reason: ", error);
```

Comment:
Yes, the initial idea was to show a particular message to the user, when we know the error better, but currently all known errors reported from the library are handled early on in the application.